packages/@aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-listener.ts

```diff
@@ -311,12 +311,7 @@ export class NetworkListener extends BaseListener implements INetworkListener {
 /**
  * Properties to reference an existing listener
  */
-export interface INetworkListener extends IResource {
-  /**
-   * ARN of the listener
-   * @attribute
-   */
-  readonly listenerArn: string;
+export interface INetworkListener extends IListener {
```

Comment:
@corymhall will these changes moving to a shared IListener interface cause the same problems as in #22567?  TBH I dont entirely grok what the issue was there

Reply:
I think we are fine in this case. We would only run into that issue if we changed the return type of something from `INetworkListener` -> `IListener` (which we are not doing).

i.e.

Before
```ts
interface {
  someProp: INetworkListener;
}

someFunc(): INetworkListener {}

someProp: someFunc(),
```

After
```ts
interface {
  someProp: INetworkListener;
}

someFunc(): IListener {} // changed!

someProp: someFunc(), // won't work!
```

TypeScript views `INetworkListener` & `IListener` as the same type because right now they are, but they _could_ be different (I guess which is why Java views them as different types).